internal/constants/scripts.go

```diff
@@ -66,140 +70,149 @@ function map_host_macs_to_interfaces() {
   for nic in $( ls $SYS_CLASS_NET_DIR )
   do
     mac=$(cat $SYS_CLASS_NET_DIR/$nic/address | tr '[:lower:]' '[:upper:]')
-    host_macs_to_hw_iface[$mac]=$nic
+    [[ -n "$mac" ]] && host_macs_to_hw_iface[$mac]=$nic
   done
 }
 
-function find_host_directory_by_mac_address() {
-  for d in $(ls -d ${NMCONNECTIONS_DIR}/host*)
-  do
+# Copy all nmstate files representing a host nic
+function copy_physical_nics_files() {
+  host_src_dir=$1
+  mapping_file=$2
+  host_work_dir=$3
+  for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]');  do
+    host_nic="${host_macs_to_hw_iface[${mac_address}]:-}"
+    if [[  -n "$host_nic" ]]; then
+      logical_nic="${logical_mac_nic_map[${mac_address}]:-}"
+      if [[ -n "$(ls -A $host_src_dir/*.nmconnection )" ]] ; then
+        for nmfile in $host_src_dir/*.nmconnection ; do
+          filename=$(basename $nmfile)
+          prefix="${filename%%.*}"
+          extension="${filename#*.}"
+          [[ -n "$prefix" ]] && [[ "$prefix" == "$logical_nic" ]] &&  cp $nmfile $host_work_dir/${host_nic}.${extension} && rm -f $nmfile
+        done
+      fi
+    fi
+  done
+}
 
-    mapping_file="${d}/${MAC_NIC_MAPPING_FILE}"
+# If any nmstate files of the current host directory were copied, copy the remaining nmstate files
+function copy_other_files() {
+  host_src_dir=$1
+  host_work_dir=$2
+  if [ "$(ls -A $host_work_dir )" ] && [[ -n "$(ls -A ${host_src_dir}/*.nmconnection)" ]] ; then
+    for nmfile in ${host_src_dir}/*.nmconnection ; do
+      filename=$(basename $nmfile)
+      target_file=$host_work_dir/$filename
+
+      # If the same filename was not used already, then use it. Otherwise generate another name.
+      if ! [[ ${allocated_nics[$filename]} ]] ; then
+        cp $nmfile $target_file && rm -f $nmfile
+      else
+        prefix="${filename%%.*}"
+        extension="${filename#*.}"
+        name=$(printf "%s-%03d" $prefix $conflicting_files_count)
+        conflicting_files_count=$(( conflicting_files_count + 1 ))
+        target_file=$host_work_dir/${name}.${extension}
+        sed  -e "s/=$prefix\$/=$name/g" -e "s/=$prefix\./=$name\./g"  $nmfile > $target_file && rm -f $nmfile
+      fi
+      allocated_nics[$filename]=true
+    done
+  fi
+}
+
+# Update references to host nics in the nmstate files
+function update_physical_nics_references() {
```

Comment:
Why do this later vs editing the file when we select it to go into the working directory in `copy_physical_nics_files`?
I don't think we'll have hardware names for virtual interfaces (bonds/vlans).

Reply:
This is how vlan connection file looks like:
```
[connection]
autoconnect=true
autoconnect-slaves=-1
id=eth1.101
interface-name=eth1.101
type=vlan
uuid=6d0f1528-fd9c-52a9-9aa5-b825aa883bc3

[ipv4]
method=disabled

[ipv6]
method=disabled

[vlan]
id=101
parent=eth1

```
As you can see, the parent hw interface is referenced here

Reply:
Oh so you're replacing both the usages in the file corresponding to that interface as well as the usage in the bond/vlans. Okay makes sense.